client/landing/stepper/declarative-flow/flows/onboarding-unified/onboarding-unified.ts

```diff
@@ -0,0 +1,155 @@
+			if ( providedDependencies.siteSlug ) {
+				return addQueryArgs( `/home/${ providedDependencies.siteSlug }`, { ref: flowName } );
+			}
+			// Fallback to generic home if no siteSlug
+			return '/';
+		};
+
+		/**
+		 * Handle step submissions for the onboarding unified flow
+		 */
+		const submit: SubmitHandler< typeof initialize > = async ( submittedStep ) => {
+			const { slug, providedDependencies } = submittedStep;
+
+			switch ( slug ) {
+				case 'plans': {
+					// Set domain origin as not set since this flow skips domain selection
+					setSignupDomainOrigin( SIGNUP_DOMAIN_ORIGIN.NOT_SET );
+
+					// Create minimal siteParams for post-checkout site creation
+					const siteParams = {
+						blog_name: '', // Will be auto-generated from username if empty
+						blog_title: translate( 'My Site' ), // Default site title
+						public: Visibility.PublicNotIndexed, // Coming soon by default
+						options: {
+							site_creation_flow: flowName, // Track which flow created the site
+							wpcom_public_coming_soon: 1, // Launch as coming soon
+						},
+						find_available_url: true, // Auto-find available URL since no domain was selected
+						validate: false,
+					};
+
+					// Save siteParams to localStorage for checkout to use
+					try {
+						window.localStorage.setItem( 'siteParams', JSON.stringify( siteParams ) );
+					} catch ( error ) {
+						// Silently fail if localStorage is not available
+					}
+
+					// Redirect directly to siteless checkout with selected plan
+					if ( providedDependencies?.cartItems?.length ) {
+						// Set completion tracking for post-checkout site creation
+						setSignupCompleteFlowName( flowName );
+
+						// Set the post-checkout destination to go through onboarding-unified flow steps
+						// Use our own flow's post-checkout step instead of borrowing from regular onboarding
+						persistSignupDestination( `/setup/${ flowName }/post-checkout-onboarding` );
+
+						// Get the selected plan from cartItems
+						const planItem = providedDependencies.cartItems.find( ( item ) => item.product_slug );
+
+						if ( planItem ) {
+							// Use dedicated onboarding-unified siteless checkout with plan in URL (similar to Jetpack/Akismet)
+							// Don't add signup=1 for logged-in users to avoid account creation conflicts
+							const urlParams = new URLSearchParams();
+							urlParams.set( 'flow', flowName );
+
+							// Only add signup=1 for logged-out users
+							if ( ! userIsLoggedIn ) {
+								urlParams.set( 'signup', '1' );
+							}
+
+							const checkoutUrl = `/checkout/unified/${
+								planItem.product_slug
+							}?${ urlParams.toString() }`;
+							return window.location.assign( checkoutUrl );
+						}
+					}
+
+					// Fallback to error if no plan selected
+					return navigate( 'error' );
+				}
+
+				case 'post-checkout-onboarding':
+					setShouldShowNotification( providedDependencies?.siteId as number );
```

Comment:
I would really appreciate it if you could type [this step](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/post-checkout/post-checkout-onboarding.tsx#L28) correctly. Like so:

```ts
const PostCheckoutOnboarding: StepType< {
	submits: {
		siteId: number;
		siteSlug: string;
	};
} >
```

This way you won't need `as number`.